scala/tempo/src/main/scala/com/databrickslabs/tempo/io.scala

```diff
@@ -0,0 +1,59 @@
+package com.databrickslabs.tempo
+
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.SparkSession
+
+
+/**
+The following methods provide support for writing the TSDF object to Delta Lake based on the partition columns specified along with the time as ZORDER columns (with Hilbert space-filling curve optimization)
+ */
+object TSDFWriters {
+  /**
+    * param: tsdf: input TSDF object to write
+    * param: tabName Delta output table name
+    * param: optimizationCols list of columns to optimize on (time)
+    */
+  def write(tsdf: TSDF, spark: SparkSession, tabName: String, optimizationCols: Option[Seq[String]] = None): Unit = {
```

Comment:
instead of passing spark in, perhaps you can create a trait `SparkSessionWrapper` spark.getOrElse and just use it to extend objects like this that way you always have access to existing spark object?

[Here's an example](https://github.com/databrickslabs/overwatch/blob/a6e7e1267d6792c935d07ca9ee99478cf5637307/src/main/scala/com/databricks/labs/overwatch/utils/SparkSessionWrapper.scala#L14) and I was able to shove in some other global goodies as well.